src/app/home/pages/PlanDebugPage/PlanDebugPage.tsx

```diff
@@ -23,23 +23,40 @@ import {
 import { IDebugRefWithStatus } from '../../../debug/duck/types';
 
 import { convertRawTreeToViewTree } from '../../../debug/duck/utils';
-import { IDebugReducerState, startDebugPolling, stopDebugPolling } from '../../../debug/duck/slice';
-import QuestionCircleIcon from '@patternfly/react-icons/dist/js/icons/question-circle-icon';
+import {
+  IDebugReducerState,
+  startDebugPolling,
+  stopDebugPolling,
+  clearJSONView,
+} from '../../../debug/duck/slice';
 import { debugSelectors } from '../../../debug/duck';
 import { IPlan } from '../../../plan/duck/types';
 import { planSelectors } from '../../../plan/duck';
 import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons/dist/js/icons/outlined-question-circle-icon';
 import { TreeContainer } from './components/TreeContainer';
 
 export const PlanDebugPage: React.FunctionComponent = () => {
-  const { planName } = useParams();
+  const { planName, migrationID } = useParams();
   const plans: IPlan[] = useSelector((state) => planSelectors.getPlansWithStatus(state));
   const dispatch = useDispatch();
   const debug: IDebugReducerState = useSelector((state) => state.debug);
   const debugRefs: IDebugRefWithStatus[] = useSelector((state) =>
     debugSelectors.getDebugRefsWithStatus(state)
   );
 
+  useEffect(() => {
+    // Start polling when Debug component loads
+    const { isPolling } = debug;
```

Comment:
Here's where we start and stop polling now. It happens on debug view component load instead of on expansion.